sechub-server/src/test/java/com/daimler/sechub/domain/schedule/ScheduleJobMarkerServiceTest.java

```diff
@@ -13,62 +13,63 @@
 
 import com.daimler.sechub.domain.schedule.job.ScheduleSecHubJob;
 import com.daimler.sechub.domain.schedule.job.SecHubJobRepository;
+import com.daimler.sechub.domain.schedule.strategy.FirstComeFirstServeSchedulerStrategy;
+import com.daimler.sechub.domain.schedule.strategy.SchedulerStrategyFactory;
 
 public class ScheduleJobMarkerServiceTest {
 
-	private SecHubJobRepository jobRepository;
+    private SecHubJobRepository jobRepository;
+    private SchedulerStrategyFactory factory;
+    private FirstComeFirstServeSchedulerStrategy strategy;
 
-	private ScheduleSecHubJob secHubJob;
+    private ScheduleSecHubJob secHubJob;
 
-	private UUID uuid;
+    private UUID uuid;
 
-	private ScheduleJobMarkerService serviceToTest;
+    private ScheduleJobMarkerService serviceToTest;
 
-	@Before
-	public void before() throws Exception {
-		serviceToTest = new ScheduleJobMarkerService();
+    @Before
+    public void before() throws Exception {
+        serviceToTest = new ScheduleJobMarkerService();
+        factory = mock(SchedulerStrategyFactory.class);
+        strategy = mock(FirstComeFirstServeSchedulerStrategy.class);
+        jobRepository = mock(SecHubJobRepository.class);
+        
+        uuid = UUID.randomUUID();
 
-		uuid = UUID.randomUUID();
+        serviceToTest.jobRepository = jobRepository;
+        serviceToTest.schedulerStrategyFactory = factory;
+        strategy.jobRepository = jobRepository;
 
-		jobRepository = mock(SecHubJobRepository.class);
+        secHubJob = mock(ScheduleSecHubJob.class);
 
-		serviceToTest.jobRepository = jobRepository;
-		
-		secHubJob = mock(ScheduleSecHubJob.class);
-		
-		when(secHubJob.getUUID()).thenReturn(uuid);
-	}
+        when(factory.build()).thenReturn(strategy);
+        when(strategy.nextJobId()).thenReturn(uuid);
+        when(jobRepository.getJob(any())).thenReturn(Optional.of(secHubJob));
```

Comment:
This changes made the test more readable ... okay.. good.
But the former test setup did insist that the fetching for a job is really doing a fetch for the given job UUID.
Using any() makes testing more easy - fair - but we reduce the scope of testing here.

Maybe you could change to 
```java
when(jobRepository.getJob(eq(uuid))).thenReturn(Optional.of(secHubJob));
```
?

Reply:
done